lib/app/models/inferno_test.rb

```diff
@@ -0,0 +1,64 @@
+module Inferno
+  module Sequence
+    class InfernoTest
+      attr_reader :name, :index, :test_block
+
+      def initialize(name, index, &test_block)
+        @name = name
+        @test_block = test_block
+        @index = index
+        load_metadata
+      end
+
+      def metadata
+        yield
+        raise MetadataException
+      end
+
+      def load_metadata
+        begin
+          instance_eval(&test_block)
+        rescue MetadataException
+        end
+      end
+
+      def id(id = nil)
+        @id ||= id
+      end
+
+      def link(link = nil)
+        @link ||= link
+      end
+
+      def optional
+        @optional = true
+      end
+
+      def optional?
+        @optional
+      end
+
+      def required?
+        !optional?
+      end
+
+      def desc(description = nil)
```

Comment:
This could be changed to `description` now to be consistent with Sequence descriptions, but it would require updating every test.

Reply:
I think I made it that way to keep the DSL a little less verbose within each test (maybe?) but I do kinda disagree with that approach now.  One thing we could do is allow either (and maybe mark it for deprecation?).  We should wait until a major version before we yank it regardless.

Reply:
I agree with @arscan here. Do we have a way to mark things for deprecation?

Reply:
I think we'd basically need to have some log output here stating that the method is deprecated.

Reply:
fhir_models has a mixin for that:

https://github.com/fhir-crucible/fhir_models/blob/master/lib/fhir_models/deprecate.rb

Not sure if it is worthwhile to handle it generically like this or not, but it is a neat solution.  Note that this was contributed by Kevin from McKesson.

Reply:
Yeah, that's nice, but I probably wouldn't bring something like that in until we have multiple deprecations.

Reply:
`caller(0..0)` doesn't provide helpful information, seems like `caller(1..1)` is more useful because it shows the sequence where this is called.  If the `Test` class had a reference back to the sequence it is a part of we could avoid system calls, which make me a little nervous for use on other platforms we don't actively test.  But if that is messy then I'm good with this.